src/logging.jl

```diff
@@ -2,6 +2,19 @@ import Logging, LoggingExtras
 
 export MPILogger, MPIFileLogger
 
+"""
+    default_logger()
+    default_logger(ctx::AbstractCommsContext)
+
+Return the default logger for the given context. If no context is passed, obtain the default context.
+
+Passing an MPICommsContext will return the `MPILogger`.
+Any other context will return the default `ConsoleLogger`.
+"""
+default_logger() = default_logger(context())
```

Comment:
Before merging this, can we sketch out how it would look like in ClimaAtmos?

My thinking is: 
- we want to save the logs in the `output_active` so that each run has its own logs
- there is an asymmetry: if you run with one process, you don't get files, if you run with more, you get files. Maybe we should try to be more uniform

Reply:
So for atmos, we want one function `default_logger` with the following behavior
- If MPI, log all procs to files, log rank 0 to stdout (MPIFileLogger)
- else, log to file and stdout.

This is easy enough, my two questions are:
- Where will `default_logger` be called? Ideally this would happen ASAP and at the top level in the driver, but then we need to duplicate the `OutputPathGenerator` calls.
- what should the defaults for the singleton file be and how will the be overridden?